lahja/base.py

```diff
@@ -211,6 +202,72 @@ def broadcast_nowait(
         """
         ...
 
+    #
+    # Subscription API
+    #
+    @abstractmethod
+    def get_connected_endpoints_and_subscriptions(
+        self
+    ) -> Tuple[Tuple[Optional[str], Set[Type[BaseEvent]]], ...]:
+        """
+        Return all connected endpoints and their event type subscriptions to this endpoint.
```

Comment:
nitpick: it's not clear what it means for a subscription to be to an endpoint. Maybe: "Return all connected endpoints and the set messages each connected endpoint is subscribed to"